final_assignment/micu_matei/installTuxyLab.py

```diff
@@ -0,0 +1,418 @@
+        write_log('e', "Nu am putut seta cwd :" + cwd)
+        write_log('e', "Probabil nu exista")
+        return False
+
+    for name in env_variables:
+        os.environ[name] = env_variables[name]
+
+    write_log('o', "Am setat variabilele de sistem ")
+    write_log('d', env_variables)
+
+    it_workerd = False
+    while attemts:
+        attemts -= 1
+        if shell:
+            try:
+                ret = os.system(command)
+            except OSError as error:
+                write_log('e', "Nu am putut rula scriptul !")
+                write_log('e', "Commanda " + command)
+                write_log('d', error)
+        else:
+            pass
+
+        if not check_exit_code:
+            it_workerd = True
+            break
+
+        if ret in check_exit_code:
+            it_workerd = True
+            write_log('o', "Am rulat scriptul :" + command)
+            write_log('o', "Am avut exitcodul " + str(ret))
+            break
+        else:
+            write_log('e', "Am rulat scriptul :" + command)
+            write_log('e', "Am avut exitcodul " + str(ret))
+            write_log('e', "Expected exitcode :")
+            write_log('d', check_exit_code)
+
+    time.sleep(int(retry_interval))
+
+    return it_workerd
+
+
+def function_delete(functie_name, valori):
+    """ deletam ce se afla la path"""
+    path_to_delete = valori.get("path", None)
+
+    if not path_to_delete:
+        write_log('e', "Nu ati dat un path")
+        return False
+
+    if not os.path.exists(path_to_delete):
+        write_log('e', "Pathul nu este valid :" + path_to_delete)
+
+    if os.path.isfile(path_to_delete):
+        os.remove(path_to_delete)
+        write_log('o', "Am eliminat fisierul :" + path_to_delete)
+
+    elif os.path.isdir(path_to_delete):
+        shutil.rmtree(path_to_delete)
+        write_log('o', "Am eliminat directorul :" + path_to_delete)
+
+    write_log('o', "Am executat cu succes " + functie_name)
+    return True
+
+
+def function_shutdown(valori):
+    """ restartam calculator """
+
+    command = "win32api call"
+    if valori['method'] == "soft":
+        command = "shutdown now"
+    elif valori['method'] == "hard":
+        command = "halt now"
+
+    # sa nu dau restart din prostie
+    command = "asdasd"
+
+    platfm = platform.system()
+
+    write_log('h', "Dam restart folosind :" + command)
+    write_log('d', platfm)
+
+    if platfm == "Linux":
+        os.system(command)
+    elif platfm == "Windows":
+        import win32api
+        win32api.InitiateSystemShutdown()
+    return True
+
+
+def create_accout_linux(name, p_grup, groups, exp_date, passwd):
+    """ cream un cont pe linux"""
+    write_log('w', "Trebuie sa adaugam un user cu :")
+    write_log('d', name)
+    write_log('d', p_grup)
+    write_log('d', groups)
+    write_log('d', exp_date)
+    write_log('d', passwd)
+
+
+def create_accout_windows(name, p_grup, groups, exp_date, passwd):
+    """ cream un cont pe linux"""
```

Comment:
Aici am ramas fara timp si am pus in log toate lucrurile ca sa nu spuna pylint si flake8 ca nu am folosit variabilele, atat tot.